core/tests/ts-integration/tests/api/web3.test.ts

```diff
@@ -932,6 +934,24 @@ describe('web3 API compatibility tests', () => {
         expect(txFromApi.v! <= 1).toEqual(true);
     });
 
+    // We want to be sure that correct(outer) contract address is return in the transaction receipt,
+    // when there is a contract that initializa another contract in the constructor
+    test('Should check inner-outer contract address in the receipt of the deploy tx', async () => {
+        const nonce = await alice.getNonce();
+        const expectedAddress = zksync.utils.createAddress(alice.address, nonce);
```

Comment:
You should use "deploy nonce", not the usual one. Tbh it will be easier if you simply check bytecode/bytecodeHash for `receipt.contractAddress` to match expected one

Reply:
1. what do you mean by "deploy nonce"?
2. why to check bytecodes if the goal is to check the fact we extract correct contract addresses from the DB?

Reply:
1. Check docs https://docs.zksync.io/build/developer-reference/ethereum-differences/nonces
2. Yes, but the correct address should be the only one with the bytecode of `outerContract`

Reply:
I see, thanks!

Reply:
feel free to check